docs/wallet-integration-guide/external-docs/party-management/create-party.rst

```diff
@@ -1,15 +1,120 @@
+
+The wallet SDK has fingerprint generation built in:
+.. code-block:: javascript
+
+   import { TopologyController } from "@splice/wallet-sdk";
+    // static method call
+    const fingerPrint = TopologyController.createFingerprintFromPublicKey(publicKey)
+
+
+
+Generating the topology transactions
+------------------------------------
+When onboarding using external signing, multiple topology transactions are required to be generated and signed in order to facilitate this.
+This is because both the keyHolder (the party) and the node (the validator) needs to agree on the hosting relationship.
+The three transactions that needs to be generated are:
+- PartyToParticipant: This transaction indicates that the party agrees to be hosted by the participant (validator).
+- ParticipantToParty: This transaction indicates that the participant (validator) agrees to host the party.
+- KeyToParty: This transaction indicates that the key (public key) is associated with the party.
+
+Once all the transactions are build they can be combined into a single hash and submitted as part of a single signature.
+The wallet SDK has helper functions to generate these transactions:
+.. code-block:: javascript
+    import { WalletSDKImpl, TopologyController } from "@splice/wallet-sdk";
+
+    // it is important to configure the SDK correctly else you might run into connectivity or authentication issues
+    const sdk = new WalletSDKImpl().configure({
+        logger: console,
+        authFactory: localAuthDefault, // or use your specific configuarion
+        ledgerFactory: localLedgerDefault, // or use your specific configuarion
+        topologyFactory: localTopologyDefault, // or use your specific configuarion
+    })
+
+    const {publicKey, privateKey } = TopologyController.createNewKeyPair()
+    //partyHint is optional but recommended to make it easier to identify the party
+    const partyHint = "my-wallet-1"
+    const preparedParty = await sdk.topology?.prepareExternalPartyTopology(publicKey, partyHint)
+
+
+preparedParty will have the following structure:
+.. code-block:: javascript
+    export type PreparedParty = {
+    partyTransactions: Uint8Array<ArrayBufferLike>[] // Array of the three topology transactions
+    combinedHash: string // sha256 hash of the three transactions that needs to be signed
+    txHashes: Buffer<ArrayBuffer>[] // Array of the three transaction hashes
+    namespace: string // the namespace of the party
+    partyId: string // the partyId as defined by ${partyHint}::${fingerprint}
+    }
+
+Sign multi-hash
+-----------------
+Since the topology transactions need to be submitted together the combined hash needs to be signed.
+The wallet SDK has a helper function to sign the combined hash:
+.. code-block:: javascript
+    import { SignTransactionHash } from "@splice/wallet-sdk";
+
+    const signature = await SignTransactionHash(preparedParty.combinedHash, privateKey)
```

Comment:
(by convention, ordinary functions in JS shouldn't start with an uppercase. but as per the other comment, we may want to expose this as `KeyController.sign(...)` or similar)